src/librustc/middle/ty/util.rs

```diff
@@ -578,6 +578,16 @@ impl<'tcx> ty::ctxt<'tcx> {
         });
         let generics = adt.type_scheme(self).generics;
 
+        // RFC 1238: if the destructor method is tagged with the
```

Comment:
Oh, yes, this code as it stands could be reduced to that.

My intention had originally been to continue returning `true` from `is_adt_dtorck` if the adt has region params.

Looking now, I see that the RFC directly says in its description of [UGEH](https://github.com/rust-lang/rfcs/blob/master/text/1238-nonparametric-dropck.md#the-new-escape-hatch-an-unsafe-attribute) that the attribute "will allow a destructor to side-step the dropck's constraints."  For some reason while writing the RFC I had thought that was only talking about the type parameters (that's why there's the explicit alternative of having [UGEH for lifetime parameters](https://github.com/rust-lang/rfcs/blob/master/text/1238-nonparametric-dropck.md#ugeh-for-lifetime-parameters)).

But at this point, I don't see much reason to stick with my original interpretation.  This one (where UGEH causes all constraints to be ignored) is easier to explain and implement.